pytest_plugins/xdist.py

```diff
@@ -33,22 +36,27 @@ def align_to_satellite(worker_id, satellite_factory):
 
     # attempt to align a worker to a satellite
     if settings.server.xdist_behavior == 'run-on-one' and settings.server.hostnames:
-        cache_proxy['server.hostname'] = settings.server.hostnames[0]
+        target_satellite_fqdn = settings.server.hostnames[0]
     elif settings.server.hostnames and worker_pos < len(settings.server.hostnames):
-        cache_proxy['server.hostname'] = settings.server.hostnames[worker_pos]
+        target_satellite_fqdn = settings.server.hostnames[worker_pos]
     elif settings.server.xdist_behavior == 'balance' and settings.server.hostnames:
-        cache_proxy['server.hostname'] = random.choice(settings.server.hostnames)
+        target_satellite_fqdn = random.choice(settings.server.hostnames)
     # get current satellite information
     elif settings.server.xdist_behavior == "on-demand":
         on_demand_sat = satellite_factory()
         if on_demand_sat.hostname:
-            cache_proxy['server.hostname'] = on_demand_sat.hostname
+            target_satellite_fqdn = on_demand_sat.hostname
         # if no satellite was received, fallback to balance
         if not settings.server.hostname:
-            cache_proxy['server.hostname'] = random.choice(settings.server.hostnames)
+            target_satellite_fqdn = random.choice(settings.server.hostnames)
+    cache_proxy['server.hostname'] = target_satellite_fqdn
     logger.info(f"xdist worker {worker_id} " f"was assigned hostname {settings.server.hostname}")
     settings.configure_airgun()
     settings.configure_nailgun()
-    yield
+
+
+@pytest.hookimpl(trylast=True)
+def pytest_sessionfinish(session, exitstatus):
+    # TODO store the deployed sat in the session so we can check it in here.
     if on_demand_sat and settings.server.auto_checkin:
```

Comment:
how is on_demand_sat in scope here? some pytest magic?

Reply:
PR in draft state, see TODO above.

Reply:
hahaha ok, just making sure I wasn't missing out on some weird pytest scope injection or something.